src/Framework/BlockManifestGenerator.php

```diff
@@ -0,0 +1,238 @@
+ * Used primarily in local development to auto-detect when blocks are added or removed.
+ */
+class BlockManifestGenerator
+{
+    /**
+     * Generate a block manifest for the given base path.
+     * Discovers all modules and their blocks, then writes sitchco.blocks.json.
+     *
+     * @param string $basePath The base directory to search (e.g., core plugin root or theme root)
+     * @return array The generated manifest data
+     */
+    public function generate(string $basePath): array
+    {
+        $basePath = FilePath::create($basePath);
+        $blocks = $this->discoverBlocks($basePath);
+
+        $manifest = [
+            'version' => BlockManifestRegistry::MANIFEST_VERSION,
+            'hash' => $this->calculateHash($blocks),
+            'blocks' => $blocks,
+        ];
+
+        $this->writeManifest($basePath, $manifest);
+
+        return $manifest;
+    }
+
+    /**
+     * Check if a manifest needs regeneration by comparing hashes.
+     * Only relevant in local environment.
+     *
+     * @param string $basePath The base directory containing the manifest
+     * @return bool True if manifest is missing or stale
+     */
+    public function shouldRegenerate(string $basePath): bool
+    {
+        $basePath = FilePath::create($basePath);
+        $manifestPath = $basePath->append(BlockManifestRegistry::MANIFEST_FILENAME);
+
+        if (!$manifestPath->exists()) {
+            return true;
+        }
+
+        try {
+            $jsonContent = file_get_contents($manifestPath->value());
+            if ($jsonContent === false) {
+                return true;
+            }
+            $existingManifest = json_decode($jsonContent, true);
+            if (json_last_error() !== JSON_ERROR_NONE || !is_array($existingManifest)) {
+                return true;
+            }
+
+            $existingHash = $existingManifest['hash'] ?? '';
+            $currentBlocks = $this->discoverBlocks($basePath);
+            $currentHash = $this->calculateHash($currentBlocks);
+
+            return $existingHash !== $currentHash;
+        } catch (\Throwable $e) {
+            error_log(
+                sprintf(
+                    'Sitchco Blocks Error: Failed to check manifest staleness at "%s". Error: %s',
+                    $manifestPath->value(),
+                    $e->getMessage(),
+                ),
+            );
+            return true;
+        }
+    }
+
+    /**
+     * Ensure manifests are fresh for all provided base paths.
+     * Checks each path and regenerates if stale.
+     *
+     * @param array<FilePath> $basePaths Array of FilePath objects to check
+     * @return void
+     */
+    public function ensureFreshManifests(array $basePaths): void
+    {
+        foreach ($basePaths as $basePath) {
+            if ($this->shouldRegenerate($basePath->value())) {
+                $this->generate($basePath->value());
+            }
```

Comment:
We should pass the FilePath instance directly to these two methods. Right now they both take a string and have to internally re-create the FilePath.